### PR TITLE
Interactive CLI improvements

### PR DESCRIPTION
**Pull request by DmitrySharabin**
This PR addressed issues discussed in #7:

- In verbose mode, don't skip child tests that have finished and everything passes
- Make the highlighted nodes more prominent
- Add support for the `expand` option—groups with failed child tests will be expanded by default

### CODE REVIEW

File: src/env/node.js [hunk 1]
```diff
@@ -41,11 +48,12 @@ function getTree (msg, i) {
 	if (msg.collapsed !== undefined) {
 		let {collapsed, highlighted, children} = msg;
 
-		let icon = msg.collapsed ? "▶︎ " : "▼ ";
-		msg = icon + msg;
+		let icon = collapsed ? "▷" : "▽";
 		if (highlighted) {
+			icon = `<c green><b>${ collapsed ? "▶︎" : "▼" }</b></c>`;
```
Comment (LeaVerou) [inline src/env/node.js:53]: These should not be magic strings randomly in the code, they should be referenced from some kind of data structure.

File: src/env/node.js [hunk 2]
```diff
@@ -139,6 +147,7 @@ Press <b>^C</b> (<b>Ctrl+C</b>) or <b>q</b> to quit interactive mode.
 			readline.emitKeypressEvents(process.stdin);
 			process.stdin.setRawMode(true); // handle keypress events instead of Node
 
+			makeCollapsible(root, {...options, expand: options?.expand ?? "fail"});
```
Comment (LeaVerou) [inline src/env/node.js:150]: I think this behavior should be the default. Also, there are no docs, so I'm not sure how `expand` is meant to be used.

File: src/env/node.js [hunk 3]
```diff
@@ -12,13 +12,20 @@ import { globSync } from 'glob';
 import format from "../format-console.js";
 import { getType, interceptConsole, restoreConsole } from '../util.js';
 
-// Recursively traverse a subtree starting from `node` and make (only) groups of tests collapsible
-function makeCollapsible (node) {
+// Recursively traverse a subtree starting from `node` and make (only) groups of tests collapsible.
```
Comment (LeaVerou) [inline src/env/node.js:15]: This should be JSDoc

**Review by LeaVerou — CHANGES_REQUESTED** [review]

File: src/env/node.js [hunk 4]
```diff
@@ -39,13 +52,21 @@ function getVisibleGroups (node, options, groups = []) {
 
 function getTree (msg, i) {
 	if (msg.collapsed !== undefined) {
+		const icons = {
+			collapsed: "▷",
+			expanded: "▽",
+			highlightedCollapsed: "▶︎",
+			highlightedExpanded: "▼",
```
Comment (LeaVerou) [inline src/env/node.js:59]: Nit: I'd put the adjective after, e.g. `collapsed_highlighted`, since it's a variation of collapsed.

File: src/env/node.js [hunk 5]
```diff
@@ -12,13 +12,26 @@ import { globSync } from 'glob';
...
+function makeCollapsible (node, options = {}) {
 	if (node.tests?.length) {
-		node.collapsed = true; // all groups are collapsed by default
+		let expand = [...new Set((options.expand ?? "fail").split(/\s+/))];
+
+		// Root is collapsed by default.
+		// All groups are collapsed by default or if the verbose option is specified.
+		// Otherwise, it will be expanded if a group contains tests, because of which
+		// the group should be expanded (e.g., failed tests).
+		node.collapsed = node.parent && !options.verbose && expand.some(o => node.stats[o] > 0) ? false : true;
```
Comment (LeaVerou) [inline src/env/node.js:31]: I’m not quite sure about this `expand` setting. What other use cases does `expand` have besides fail? How would you disable the auto-expansion for fail?

Also, perhaps you could split that into a separate PR so we can merge the formatting changes asap.

File: src/env/node.js (hunk 4 above)
Comment (DmitrySharabin) [reply src/env/node.js:59]: Agreed! Thanks.

**Comment by LeaVerou** [comment]
So I think it would make sense to code this behavior as the default behavior, and if we get any use cases for a different behavior we can examine what the best API would be for that. We cannot design an API without a good understanding of use cases.

**Comment by DmitrySharabin** [comment]
Look at this case I stumbled on today—if we automatically expand all groups with failed tests, we might end up with ruined output. The new failed test on `Color.js` is an excellent example.

https://github.com/LeaVerou/htest/assets/9166277/5fec7871-61e8-41b9-aed3-c7939953af66

We probably should also think of the output of a test's results.

**Comment by LeaVerou** [comment]
While we certainly shouldn't break so badly in that case, I do wonder *why* that test output there is so long...

**Comment by DmitrySharabin** [comment]
This PR was divided into several smaller (already merged) PRs, so we don't need it anymore. I'm going to close it.
